File: helix/src/main/java/glitch/helix/object/json/list/Streams.java
package glitch.helix.object.json.list;

import com.google.common.collect.ImmutableList;
import com.google.gson.annotations.SerializedName;
import glitch.api.objects.json.interfaces.CursorList;
import glitch.helix.object.json.Stream;
import lombok.Data;

@Data
public class Streams implements CursorList<Stream> {
    private final ImmutableList<Stream> data;

    @SerializedName("pagination.cursor")
    private final String cursor;
}

File: helix/src/main/kotlin/glitch/helix/object/json/streams.kt
package glitch.helix.`object`.json

import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName
import glitch.api.objects.json.interfaces.IDObject
import glitch.helix.`object`.adapters.StreamTypeSerializer
import java.time.Instant
import java.util.*

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class Stream(
        override val id: Long,
        val userId: Long,
        val userName: String,
        val gameId: Long,
        val communityIds: Array<String>,
        @SerializedName("type")
        @JsonAdapter(StreamTypeSerializer::class)
        val isLive: Boolean,
        val title: String,
        val viewerCount: Int,
        val startedAt: Instant,
        val language: Locale,
        val thumbnailUrl: String
) : IDObject<Long>